code/script.js

```diff
@@ -227,99 +229,146 @@ const setSecret = () => {
+        );
+        charactersInPlay = charactersInPlay.filter((person) =>
+          person[category].includes(value)
+        );
+      } else {
+        alert(
+          `No, the person doesn't wear ${value}! Remove all people that wears ${value}`
+        );
+        charactersInPlay = charactersInPlay.filter(
+          (person) => !person[category].includes(value)
+        );
+      }
+    } else if (category === "other") {
+      if (keep) {
+        alert(`Yes, the person ${value}! Keep all people that ${value}`);
+        charactersInPlay = charactersInPlay.filter((person) =>
+          person[category].includes(value)
+        );
+      } else {
+        alert(`No, the person does not ${value}! Remove all people who ${value}`);
+        charactersInPlay = charactersInPlay.filter(
+          (person) => person[category].includes(value)
+        );
+      }
+
 
-  // Determine what is the category
-  // filter by category to keep or remove based on the keep variable.
-  /* 
-    for hair and eyes :
-      charactersInPlay = charactersInPlay.filter((person) => person[attribute] === value)
-      or
-      charactersInPlay = charactersInPlay.filter((person) => person[attribute] !== value)
+    } else if (category === "hair") {
+      if (keep) {
+        alert(
+          `Yes, the person has ${value} hair! Keep all people with ${value} hair`
+        );
+        charactersInPlay = charactersInPlay.filter(
+          (person) => person[category] === value
+        );
+    
+      } else {
+        alert(
+          `No, the person does not have ${value} hair! Remove all people with ${value} hair`
+        );
+        charactersInPlay = charactersInPlay.filter(
+          (person) => person[category] !== value
+        );
+      }
 
-    for accessories and other
-      charactersInPlay = charactersInPlay.filter((person) => person[category].includes(value))
-      or
-      charactersInPlay = charactersInPlay.filter((person) => !person[category].includes(value))
-  */
 
-  // Invoke a function to redraw the board with the remaining people.
+    } else {
+      if (keep) {
+        alert(
+          `Yes, the person has ${value} eyes! Keep all people with ${value} eyes`
+        );
+        charactersInPlay = charactersInPlay.filter(
+          (person) => person[category] === value
+        );
+      } else {
+        alert(
+          `No, the person does not have ${value} eyes! Remove all people with ${value} eyes`
+        );
+        charactersInPlay = charactersInPlay.filter(
+          (person) => person[category] !== value
+        );
+      }
+    }
+ generateBoard()
 }
 
 // when clicking guess, the player first have to confirm that they want to make a guess.
 const guess = (personToConfirm) => {
-  // store the interaction from the player in a variable.
-  // remember the confirm() ?
-  // If the player wants to guess, invoke the checkMyGuess function.
+  confirm(`Do you really want to guess?`) 
+  if (true){
```

Comment:
This line is always going to be true regardless of the user choice. So, if the user wants to guess but then, the user for some reason changes his/her mind and press the cancel button, the message should not be shown and instead, the board game. 
So you need only an if statement, no need an else, and only if the user wants to guess, the message will be shown, if not, the board game. 
I am going to copy here my code so you can understand what I mean 😊

```
let confirmation = confirm('Are you really sure you want to guess on this person?'); 
  if(confirmation) {
    checkMyGuess(personToConfirm)
  }  
};
```
// The variable confirmation returns a boolean value, either true if the user accepts or false if the user doesn't want to guess. (It works the same way as the prompt(), whatever the user types, will be the value )
// If the player wants to guess(if confirmation === true), the checkMyGuess function will be invoked.